### ISSUE TITLE
Branch for 6.2.x appears to have been merged with 6.3.x via fast-forward merge with main branch

### ISSUE THREAD

**Issue by dforrest-es** [spring-projects/spring-security#15016]
**Describe the bug**
I've been keeping an eye on this repository for SAML based changes & noticed odd branch updates on `6.2.x` since my last check around April 18th. At the time, the diff stats between `6.2.x` & `main` were 415 commits, 310 files changed, & +29771 -1227 as seen [here](https://github.com/spring-projects/spring-security/compare/599cdb84bc345169652896d1ec115d0a5534814d...8397dfe3b2d8444ecb583cbbf5556743c970cfdd). Currently (May 7th) it is 45 commits, 28 files changed, & +643 -176 as seen [here](https://github.com/spring-projects/spring-security/compare/6.2.x...main).

It can also be seen that the 6.2.x branch has 6.3.x changes:
* `spring-security-6.3.rnc` & `spring-security-6.3.xsd` in config: https://github.com/spring-projects/spring-security/tree/6.2.x/config/src/main/resources/org/springframework/security/config
* `6.3.0-SNAPSHOT` version in `gradle.properties`: https://github.com/spring-projects/spring-security/blob/6.2.x/gradle.properties

An illustration below shows the `6.2.x` branch (pink line) becoming the same as the `main` branch (purple line) for a period of time between April 21st & 23rd. Of note, orange & teal lines are `6.1.x` & `5.8.x` respectively.
<img width="525" alt="broken-6 2 x-branch" src="https://github.com/spring-projects/spring-security/assets/18491489/2177b879-4a5d-4f21-8050-a7e36fb45d12">

I assume this was a mistake & has just not been noticed yet. For reference, the commit hashes for 'New 6.2.x branch' is [c33afce](https://github.com/spring-projects/spring-security/commit/c33afcebd5cee33baef0c433b6f0373001e1cd17) & for 'Old 6.2.x branch' is [7007ce3](https://github.com/spring-projects/spring-security/commit/7007ce38e116ab73d3862c35e5e22e27429285ac).

**To Reproduce**
None

**Expected behavior**
None

**Comment by marcusdacoregio** [comment]
Thank you very much for your report @dforrest-es, that indeed wasn't noticed by the team.

Thanks to the help of @sjohnr, we have now (hopefully) fixed the issue via https://github.com/spring-projects/spring-security/commit/08f11f06abe6f922e07cc7ab277b71083aa42a8c

Basically, what we did (for future reference) was hard reset the `6.2.x` branch to https://github.com/spring-projects/spring-security/commit/7007ce38e116ab73d3862c35e5e22e27429285ac, then soft reset the branch back to the latest commit. Then we reapplied every 6.2.x commit after the fast-forward happened.

We will look for measures to prevent this from happening again.
